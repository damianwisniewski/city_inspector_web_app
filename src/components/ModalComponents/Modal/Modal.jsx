import React from 'react'
import PropTypes from 'prop-types'

import './Modal.scss'
import CloseButton from '../CloseButton/CloseButton'
import Overlay from '../Overlay/Overlay'

const Modal = ({ darkOverlay, closeButton, onButtonClose, onOverlayClick, message, children }) => {
	return (
		<Overlay onOverlayClick={onOverlayClick} dark={darkOverlay || false}>
			<div className='Modal-container'>
				{message && <p className='Modal-container__extra-message'>{message}</p>}
				{closeButton && <CloseButton onClose={onButtonClose} />}
				{children}
			</div>
		</Overlay>
	)
}

Modal.propTypes = {
	darkOverlay: PropTypes.bool,
	children: PropTypes.element.isRequired,
	closeButton: PropTypes.bool,
	onClose: PropTypes.func,
}

export default Modal
