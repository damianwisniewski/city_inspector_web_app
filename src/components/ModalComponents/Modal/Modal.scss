@import '../../../assets/styles/index';

.Modal-container {
	display: flex;
	flex-direction: column;
	position: relative;
	min-width: 290px;
	max-width: calc(100% - 26px);
	min-height: 200px;
	max-height: calc(100% - 26px);
	align-content: stretch;
	background-color: $white;
	border-radius: 5px;
	box-shadow: $shadow;
	padding: 15px;
	margin: 13px;
	animation: 0.5s 1 forwards resize;

	& > *:not(button) {
		min-width: 100%;
		min-height: 100%;
		border-radius: 5px;
		overflow-x: hidden;
		overflow-y: auto;
	}

	&__extra-message {
		font-size: 12px;
		background-color: $red;
		color: $white;
		padding: 5px;
		width: 250px;
		text-align: center;
	}
}

@keyframes resize {
	0% {
		transform: scale(0);
	}

	70% {
		transform: scale(1.1);
	}

	100% {
		transform: scale(1);
	}
}
