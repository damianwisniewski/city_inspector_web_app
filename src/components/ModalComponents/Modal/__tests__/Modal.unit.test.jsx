import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Modal from '../Modal'

configure({ adapter: new Adapter() })

describe('<NavItem />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let ModalComponent
	const children = <div />

	beforeAll(() => {
		ModalComponent = shallow(<Modal>{children}</Modal>)
	})

	test('should render correctly without specyfi any props', () => {
		expect(toJson(ModalComponent)).toMatchSnapshot()
	})

	test('should pass props for overlay to Overlay component', () => {
		ModalComponent.setProps({
			darkOverlay: true,
			onOverlayClick: () => {},
		})

		expect(toJson(ModalComponent)).toMatchSnapshot()
	})

	test('should render also CloseButton for property closeButton=true', () => {
		ModalComponent.setProps({
			closeButton: true,
			onButtonClose: () => {},
		})

		expect(toJson(ModalComponent)).toMatchSnapshot()
	})
})
