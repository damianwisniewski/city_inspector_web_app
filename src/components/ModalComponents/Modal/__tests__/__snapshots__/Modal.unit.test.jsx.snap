// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> should pass props for overlay to Overlay component 1`] = `
<Overlay
  dark={true}
  onOverlayClick={[Function]}
>
  <div
    className="Modal-container"
  >
    <div />
  </div>
</Overlay>
`;

exports[`<NavItem /> should render also CloseButton for property closeButton=true 1`] = `
<Overlay
  dark={true}
  onOverlayClick={[Function]}
>
  <div
    className="Modal-container"
  >
    <CloseButton
      onClose={[Function]}
    />
    <div />
  </div>
</Overlay>
`;

exports[`<NavItem /> should render correctly without specyfi any props 1`] = `
<Overlay
  dark={false}
  onOverlayClick={[Function]}
>
  <div
    className="Modal-container"
  >
    <div />
  </div>
</Overlay>
`;
